File: app/src/main/java/com/testapp/ramboostergfxtool/adapter/WhiteListAdapter.java
package com.testapp.ramboostergfxtool.adapter;

import android.view.View;
import android.view.ViewGroup;
import androidx.annotation.Nullable;
import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.BaseViewHolder;
import com.testapp.ramboostergfxtool.R;
import com.testapp.ramboostergfxtool.util.AppSelectInfo;
import java.util.List;

public class WhiteListAdapter extends BaseQuickAdapter<AppSelectInfo, BaseViewHolder> {
    public WhiteListAdapter(@Nullable List<AppSelectInfo> list) {
        super(R.layout.item_white_app, list);
    }

    
    @Override
    public View getItemView(int i, ViewGroup viewGroup) {
        return super.getItemView(i, viewGroup);
    }

    
    public void convert(BaseViewHolder baseViewHolder, AppSelectInfo appSelectInfo) {
        onBaseView(baseViewHolder, appSelectInfo);
    }

    
    public void onBaseView(BaseViewHolder baseViewHolder, AppSelectInfo appSelectInfo) {
        baseViewHolder.setVisible(R.id.add_app_layout_content, true);
        baseViewHolder.setImageDrawable(R.id.image_icon, appSelectInfo.myAppInfo.drawable);
        baseViewHolder.setText(R.id.text_app_name, appSelectInfo.myAppInfo.charSequence);
        baseViewHolder.addOnClickListener(R.id.layout_add_root);
        baseViewHolder.setImageResource(R.id.img_add_app, appSelectInfo.aBoolean ? R.drawable.ic_switch_button_on : R.drawable.ic_switch_button_off);
    }
}

File: app/src/main/java/com/testapp/ramboostergfxtool/adapter/WhitelistAdapter.kt
package com.testapp.ramboostergfxtool.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.RelativeLayout
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import androidx.recyclerview.widget.RecyclerView.ViewHolder
import com.testapp.ramboostergfxtool.R
import com.testapp.ramboostergfxtool.activity.WhitelistActivity
import com.testapp.ramboostergfxtool.util.AppList
import com.testapp.ramboostergfxtool.util.AppSelectInfo
import com.testapp.ramboostergfxtool.util.ContactList

class WhitelistAdapter<T>(
    private var data: List<T>,
    private val onAppClicked: ((String, Boolean) -> Unit)? = null,
    private val onContactClicked: ((ContactList) -> Unit)? = null,
    private val type: String
) : RecyclerView.Adapter<WhitelistAdapter.WhiteListViewHolder>() {

    class WhiteListViewHolder(val view: View) : ViewHolder(view) {
        val imageIcon: ImageView by lazy { view.findViewById(R.id.image_icon) }
        val imgAddApp: ImageView by lazy { view.findViewById(R.id.img_add_app) }
        val textAppName: TextView by lazy { view.findViewById(R.id.text_app_name) }
        val addAppLayoutContent: RelativeLayout by lazy { view.findViewById(R.id.add_app_layout_content) }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int) = WhiteListViewHolder(
        LayoutInflater.from(parent.context).inflate(R.layout.item_white_app, parent, false)
    )

    override fun getItemCount() = data.size

    override fun onBindViewHolder(holder: WhiteListViewHolder, position: Int) {
        when (type) {
            WhitelistActivity.KEY_CLEAR_BACKGROUND_WHITELIST -> {
                (data[position] as AppSelectInfo).let { app ->
                    holder.apply {
                        imageIcon.setImageDrawable(app.myAppInfo.drawable)
                        textAppName.text = app.myAppInfo.charSequence
                        imgAddApp.setImageResource(if (app.aBoolean) R.drawable.ic_switch_button_on else R.drawable.ic_switch_button_off)
                        addAppLayoutContent.setOnClickListener {
                            app.aBoolean = app.aBoolean.not()
                            imgAddApp.setImageResource(if (app.aBoolean) R.drawable.ic_switch_button_on else R.drawable.ic_switch_button_off)
                            onAppClicked?.invoke(app.myAppInfo.string, app.aBoolean)
                        }
                    }
                }
            }
            WhitelistActivity.KEY_REJECT_CALL_WHITELIST -> {
                (data[position] as ContactList).let { contact ->
                    holder.apply {
                        imageIcon.setImageResource(R.drawable.ic_user)
                        textAppName.text = contact.name
                        imgAddApp.setImageResource(if (contact.isAdded) R.drawable.ic_switch_button_on else R.drawable.ic_switch_button_off)
                        addAppLayoutContent.setOnClickListener {
                            contact.isAdded = contact.isAdded.not()
                            imgAddApp.setImageResource(if (contact.isAdded) R.drawable.ic_switch_button_on else R.drawable.ic_switch_button_off)
                            onContactClicked?.invoke(contact)
                        }
                    }
                }
            }
            WhitelistActivity.KEY_NOTIFICATION_BLOCKING_WHITELIST -> {
                (data[position] as AppList).let { app ->
                    holder.apply {
                        imageIcon.setImageDrawable(app.icon)
                        textAppName.text = app.name
                        imgAddApp.setImageResource(if (app.isAdded) R.drawable.ic_switch_button_on else R.drawable.ic_switch_button_off)
                        addAppLayoutContent.setOnClickListener {
                            app.isAdded = app.isAdded.not()
                            imgAddApp.setImageResource(if (app.isAdded) R.drawable.ic_switch_button_on else R.drawable.ic_switch_button_off)
                            onAppClicked?.invoke(app.pkgName, app.isAdded)
                        }
                    }
                }
            }
        }
    }

    fun updateData(data: List<T>) {
        this.data = data
        notifyDataSetChanged()
    }
}